src/org/openj9/envInfo/MachineInfo.java

```diff
@@ -201,10 +202,31 @@ private void getSysInfo() {
 		putInfo(new Info("sysArch", SYS_ARCH_CMD, ce.execute(SYS_ARCH_CMD), null));
 		putInfo(new Info("procArch", PROC_ARCH_CMD, ce.execute(PROC_ARCH_CMD), null));
 		String microArchOutput = ce.execute(MICRO_ARCH_CMD);
+		String microArch = "";
+		// Check for specific machine versions and set the microArch accordingly
 		if (microArchOutput.toLowerCase().contains("skylake")) {
-			String microArch = "skylake";
+			microArch = "skylake";
+		} else if (microArchOutput.contains("8562")) {
+			microArch = "IBM z15 T02";
+		} else if (microArchOutput.contains("8561")) {
+			microArch = "IBM z15 T01";
+		} else if (microArchOutput.contains("3907")) {
+			microArch = "IBM z14 ZR1";
+		} else if (microArchOutput.contains("3906")) {
+			microArch = "IBM z14";
+		} else if (microArchOutput.contains("2965")) {
+			microArch = "IBM z13s";
+		} else if (microArchOutput.contains("2964")) {
+			microArch= "IBM z13";
```

Comment:
We do not need to know the sub-processor types. That means, z15 T02 and z15 T01 are z15. z14 ZR1 is z14. z13s is z13.